### PR TITLE
[SPARK-35928][BUILD] Upgrade ASM to 9.1

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to upgrade ASM to 9.1

### Why are the changes needed?

The latest `xbean-asm9-shaded` is built with ASM 9.1.

- https://mvnrepository.com/artifact/org.apache.xbean/xbean-asm9-shaded/4.20
- https://github.com/apache/geronimo-xbean/blob/5e0e3c0c6463f5b1e29185bdd9c2366c38371aa0/pom.xml#L67

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Pass the CIs.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Could you review this please, @srowen and @viirya ?

File: pom.xml [hunk 1]
```diff
@@ -2858,6 +2864,18 @@
...
+              <groupId>org.ow2.asm</groupId>
+              <artifactId>asm</artifactId>
+              <version>9.1</version>
+            </dependency>
+            <dependency>
+              <groupId>org.ow2.asm</groupId>
+              <artifactId>asm-commons</artifactId>
+              <version>9.1</version>
+            </dependency>
+          </dependencies>
```
Comment (viirya) [inline pom.xml:2878]: why need this?

**Review by viirya — COMMENTED** [review]
looks okay. one question.

File: pom.xml (hunk 1 above)
Comment (dongjoon-hyun) [reply pom.xml:2878]: Yes, Maven plugin is using `asm 8.0`. So, we need to match with our version.
- https://mvnrepository.com/artifact/org.apache.maven.plugins/maven-shade-plugin/3.2.4

**Comment by dongjoon-hyun** [comment]
Thank you, @viirya .
Merged to master.

**Comment by srowen** [comment]
I'm probably missing something - we don't have / need Jenkins tests, just the Github Actions? I just couldn't see test results here, or for the Chill change.

**Comment by viirya** [comment]
Seems Jenkins was not working before. (it works now)

@dongjoon-hyun @srowen I think we can watch Jenkins dashboard closely to make sure this pass Jenkins.

**Comment by dongjoon-hyun** [comment]
Thank you for review, @srowen . Yes, during the weekend, Jenkins was down.

For GitHub Action result, you can see the following at the last commit.
- https://github.com/dongjoon-hyun/spark/actions/runs/981869239

For the Jenkins result, I'll monitor as @viirya suggested.

**Comment by viirya** [comment]
Thank you @dongjoon-hyun.

**Comment by dongjoon-hyun** [comment]
Here is some update. Although the Jenkins status is super noisy due to the timeout, we got the green light for the following at least.
- Maven with Hadoop 2.7/Java11: https://amplab.cs.berkeley.edu/jenkins/view/Spark%20QA%20Test%20(Dashboard)/job/spark-master-test-maven-hadoop-2.7-jdk-11/
- SBT with Hadoop 3.2: https://amplab.cs.berkeley.edu/jenkins/view/Spark%20QA%20Test%20(Dashboard)/job/spark-master-test-sbt-hadoop-3.2/

I'll update here when I collect more info.

**Pull request closed without merging**